docs: correct type list and fix typos in type-specific keywords page #1952 (#1954)
json-schema-org/website: pages/understanding-json-schema/reference/type.md

Reviewer: "numeric types" is not a type.

```suggestion
- [integer](../../understanding-json-schema/reference/numeric)
- [number](../../understanding-json-schema/reference/numeric)
```

> 1. **A single string**. When it is a single string, it must be one of the types mentioned above (`array`, `boolean`, `integer`, `number`, `null`, `object`, `regular expressions`, or `string`). This specifies that the instance data is only valid when it matches that specific type. 

Reviewer: "regular expressions" is not a type.

```suggestion
1. **A single string**. When it is a single string, it must be one of the types mentioned above (`array`, `boolean`, `integer`, `number`, `null`, `object`, or `string`). This specifies that the instance data is only valid when it matches that specific type. 
```

> Here is an example of using the `number` keyword as a single string:

Reviewer: We usually only refer to the property names as "keywords", not their values. So, `type` is a keyword and `"number"` is the value of that keyword.

```suggestion
Here is an example of using `"number"` as a single string value:
```
